### PR TITLE
Generate code to OUT_DIR/protocol.rs instead

### PR DESCRIPTION
**Pull request by vvilhonen**
Crates should generate the code there as described [here](https://doc.rust-lang.org/cargo/reference/build-script-examples.html#code-generation). Will help to fix the docs.rs build error [here](https://docs.rs/crate/headless_chrome/1.0.0/builds/708635).

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -14,28 +16,33 @@ use crate::compile::compile_cdp_json;
 
 pub fn init() {
     const CDP_COMMIT: &str = "15f524c8f5ce5b317ddcdf5e6f875d6eb8bdac88";
+    let out_dir = env::var_os("OUT_DIR").unwrap();
     let mut file = OpenOptions::new()
         .append(true)
         .create(true)
-        .open(format!("./src/protocol.rs"))
+        .open(Path::new(&out_dir).join("protocol.rs"))
```
Comment (mdrokz) [inline src/lib.rs]: wouldnt this code fail in normal compilations since OUT_DIR variable wont be set, correct me if im wrong

Comment (vvilhonen) [reply src/lib.rs]: It's set by cargo when building so it's always present. I missed the update on rustfmt below this, adding it in a second.

**Comment by vvilhonen** [comment]
Should be good now, I also replaced reqwest with ureq so it doesn't pull the whole crates.io for a single blocking http request. What do you think?

File: src/lib.rs (hunk 1 above)
Comment (vvilhonen) [reply src/lib.rs]: Here's the documentation for the OUT_DIR https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts

**Comment by vvilhonen** [comment]
Docs are now updated with usage instructions as well

File: src/lib.rs (hunk 1 above)
Comment (mdrokz) [reply src/lib.rs]: > It's set by cargo when building so it's always present. I missed the update on rustfmt below this, adding it in a second.

no i mean when the user builds this crate that env variable wont be set

Comment (vvilhonen) [reply src/lib.rs]: When you run `cargo build` the build.rs gets ran. Before executing, cargo sets the OUT_DIR environment variable as described in the link above.

Comment (mdrokz) [reply src/lib.rs]: > When you run `cargo build` the build.rs gets ran. Before executing, cargo sets the OUT_DIR environment variable as described in the link above.

ah i understand now my brain didnt get it for some reason

**Review by mdrokz — APPROVED** [review]

**Comment by mdrokz** [comment]
> Should be good now, I also replaced reqwest with ureq so it doesn't pull the whole crates.io for a single blocking http request. What do you think?

nice good thinking

**Comment by mdrokz** [comment]
this looks good, thanks for fixing this

File: src/lib.rs (hunk 1 above)
Comment (vvilhonen) [reply src/lib.rs]: No worries, I'm on holiday mode as well but have been banging my head against the build script thingy so much recently that it still seems somewhat clear :D

**Pull request merged**